administrator/components/com_media/Model/Api.php

```diff
@@ -120,20 +121,27 @@ public function getFile($adapter, $path = '/')
 	 * @param   string  $adapter  The adapter
 	 * @param   string  $path     The folder
 	 * @param   string  $filter   The filter
+	 * @param   array   $options  The options
 	 *
 	 * @return  \stdClass[]
 	 *
 	 * @since   __DEPLOY_VERSION__
 	 * @throws  \Exception
 	 * @see     AdapterInterface::getFile()
 	 */
-	public function getFiles($adapter, $path = '/', $filter = '')
+	public function getFiles($adapter, $path = '/', $filter = '', $options = array())
 	{
-		// Add adapter info to all the files to be returned
+		// Add adapter prefix to all the files to be returned
 		$files = $this->getAdapter($adapter)->getFiles($path, $filter);
 
 		foreach ($files as $file)
 		{
+			// If requested add options
+			if (isset($options['url']) && $options['url'] == 'true' && $file->type == 'file')
```

Comment:
I would not test if the string is `'true'`. I would just do `if (isset($options['url']) && $options['url'] && $file->type == 'file')`.